service/src/main/java/greencity/service/EventCommentServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+import greencity.dto.user.UserVO;
+import greencity.entity.User;
+import greencity.entity.event.Event;
+import greencity.entity.event.EventComment;
+import greencity.exception.exceptions.NotFoundException;
+import greencity.repository.EventCommentRepo;
+import greencity.repository.EventRepo;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class EventCommentServiceImpl implements EventCommentService {
+    private EventCommentRepo eventCommentRepo;
+    private EventService eventService;
+    private ModelMapper modelMapper;
+    private final EventRepo eventRepo;
+
+    /**
+     * Method to save {@link greencity.entity.event.EventComment}.
+     *
+     * @param eventId                   id of {@link greencity.entity.event.Event}
+     *                                  to which we save comment.
+     * @param addEventCommentDtoRequest dto with
+     *                                  {@link greencity.entity.event.EventComment}
+     *                                  text, parentCommentId.
+     * @param userVO                    {@link User} that saves the comment.
+     * @return {@link AddEventCommentDtoResponse} instance.
+     */
+
+    @Override
+    public AddEventCommentDtoResponse save(Long eventId, AddEventCommentDtoRequest addEventCommentDtoRequest,
+        UserVO userVO) {
+        EventVO eventVO = eventService.findById(eventId);
+        EventComment eventComment = modelMapper.map(addEventCommentDtoRequest, EventComment.class);
+        eventComment.setUser(modelMapper.map(userVO, User.class));
+        eventComment.setEvent(modelMapper.map(eventVO, Event.class));
+        return modelMapper.map(eventCommentRepo.save(eventComment), AddEventCommentDtoResponse.class);
+    }
+
+    /**
+     * Method to count not deleted comments to certain {@link Event}.
+     *
+     * @param eventId to specify {@link Event}
+     * @return amount of comments
+     */
+    @Override
+    public int countComments(Long eventId) {
+        return eventCommentRepo.countEventCommentsByEvent(eventRepo.findById(eventId)
+            .orElseThrow(() -> new NotFoundException(ErrorMessage.EVENT_NOT_FOUND_BY_ID + eventId)));
+    }
+
+    /**
+     * Method to get all active comments to {@link Event} specified by eventId.
+     *
+     * @param pageable page of event.
+     * @param eventId  specifies {@link Event} to which we search for comments
+     * @return all active comments to certain event specified by eventId.
+     */
+    @Override
+    public PageableDto<EventCommentDto> getAllActiveComments(Pageable pageable, UserVO userVO, Long eventId) {
+        Optional<Event> event = eventRepo.findById(eventId);
```

Comment:
```suggestion
       Event event = eventRepo.findById(eventId).orElseThrow(() -> new NotFoundException(ErrorMessage.EVENT_NOT_FOUND_BY_ID + eventId));
```
With this you can remove next lines